README.md

```diff
@@ -1 +1,149 @@
+# Factorio Factory Planner
 
+A sophisticated tool for automatically generating efficient Factorio factory layouts based on your unlocked research and production goals.
+
+## Overview
+
+The Factorio Factory Planner takes your current research progress and production requirements as input, then generates optimized factory designs that maximize efficiency while minimizing resource waste and space usage. The tool outputs visual factory layouts that can be directly implemented in-game.
+
+## Goals and Features
+
+### Core Features
+- **Research-Aware Planning**: Analyzes your unlocked technologies to determine available recipes and production methods
+- **Efficient Layout Generation**: Creates optimized factory designs with minimal belt congestion and resource waste
+- **Visual Output**: Generates clear, implementable factory blueprints and images
+- **Production Goal Optimization**: Calculates optimal production ratios for target items per minute/second
+
+### Advanced Features (Planned)
+- **Modular Factory Design**: Creates reusable factory modules that can be expanded
+- **Multi-Product Optimization**: Handles complex production chains with multiple end products
+- **Space Efficiency Analysis**: Optimizes for compact designs when space is limited
+- **Power and Infrastructure Planning**: Includes power pole and logistics infrastructure in designs
+- **Blueprint Export**: Direct integration with Factorio's blueprint system
+
+## Technology Stack Options
+
+### Backend Options
```

Comment:
Remove backend options, this will be a CLI program written in Go which will generate a PNG as well as a Factorio blueprint string